ResearchKit/ActiveTasks/ORKGoNoGoContentView.m

```diff
@@ -0,0 +1,135 @@
+}
+
+- (void)startFailureAnimationWithDuration:(NSTimeInterval)duration completion:(void (^)(void))completion {
+    [_stimulusView startFailureAnimationWithDuration:duration completion:completion];
+}
+
+- (void)resetAfterDelay:(NSTimeInterval)delay completion:(nullable void (^)(void))completion {
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(delay * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        _stimulusView.hidden = YES;
+        [_stimulusView setColor:self->_stimulusColor];
+        
+        if (completion) {
+            completion();
+        }
+    });
+}
+
+- (void)addStimulusView {
+    if (!_stimulusView) {
+        _stimulusView = [[ORKGoNoGoStimulusView alloc] initWithBackgroundColor:self->_stimulusColor];
+        _stimulusView.translatesAutoresizingMaskIntoConstraints = NO;
+        _stimulusView.backgroundColor = self->_stimulusColor;
+        [self addSubview:_stimulusView];
+        [self setUpStimulusViewConstraints];
+    }
+}
+
+
+- (BOOL)stimulusHidden {
+    return _stimulusView.hidden;
+}
+
+- (void)setStimulusHidden:(BOOL)hidden {
+    _stimulusView.hidden = hidden;
+}
+
+- (void)setUpStimulusViewConstraints {
+    NSMutableArray *constraints = [NSMutableArray array];
+    
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:_stimulusView
+                                                        attribute:NSLayoutAttributeCenterX
+                                                        relatedBy:NSLayoutRelationEqual
+                                                           toItem:self
+                                                        attribute:NSLayoutAttributeCenterX
+                                                       multiplier:1.0
+                                                         constant:0.0]];
+    
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:_stimulusView
+                                                        attribute:NSLayoutAttributeTop
+                                                        relatedBy:NSLayoutRelationEqual
+                                                           toItem:self
+                                                        attribute:NSLayoutAttributeTop
+                                                       multiplier:1.0
+                                                         constant:8.0]];
+    
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"V:[_stimulusView]-(>=0)-|"
+                                                                             options:NSLayoutFormatAlignAllCenterX
+                                                                             metrics:nil
+                                                                               views:NSDictionaryOfVariableBindings(_stimulusView)]];
+    
+    [NSLayoutConstraint activateConstraints:constraints];
+}
+
+
+- (void)changeColor:(UIColor*)color {
```

Comment:
why not use a property in the header:

````
@property (nonatomic, copy) UIColor *stimulusColor;
````

And then you don't need this method?